### ISSUE TITLE
Planning failure when effective predicate contains NaN

### ISSUE THREAD

**Issue by martint** [prestosql/presto#4119]
```sql
SELECT * FROM (VALUES nan()) t(x) JOIN (VALUES 1e0) u(x) ON t.x = u.x;
```

fails with:

```
Query 20200621_020702_00004_mbps9 failed: cannot use Double NaN as range bound
java.lang.IllegalArgumentException: cannot use Double NaN as range bound
	at io.prestosql.spi.predicate.Marker.<init>(Marker.java:80)
	at io.prestosql.spi.predicate.Marker.create(Marker.java:89)
	at io.prestosql.spi.predicate.Marker.exactly(Marker.java:115)
	at io.prestosql.spi.predicate.Range.equal(Range.java:84)
	at io.prestosql.spi.predicate.SortedRangeSet.of(SortedRangeSet.java:80)
	at io.prestosql.spi.predicate.ValueSet.of(ValueSet.java:60)
	at io.prestosql.spi.predicate.Domain.singleValue(Domain.java:83)
	at io.prestosql.spi.predicate.Domain.multipleValues(Domain.java:92)
	at io.prestosql.sql.planner.EffectivePredicateExtractor$Visitor.visitValues(EffectivePredicateExtractor.java:371)
	at io.prestosql.sql.planner.EffectivePredicateExtractor$Visitor.visitValues(EffectivePredicateExtractor.java:106)
	at io.prestosql.sql.planner.plan.ValuesNode.accept(ValuesNode.java:73)
	at io.prestosql.sql.planner.EffectivePredicateExtractor.extract(EffectivePredicateExtractor.java:103)
	at io.prestosql.sql.planner.optimizations.PredicatePushDown$Rewriter.visitJoin(PredicatePushDown.java:400)
	at io.prestosql.sql.planner.optimizations.PredicatePushDown$Rewriter.visitJoin(PredicatePushDown.java:135)
	at io.prestosql.sql.planner.plan.JoinNode.accept(JoinNode.java:320)
	at io.prestosql.sql.planner.plan.SimplePlanRewriter$RewriteContext.rewrite(SimplePlanRewriter.java:84)
	at io.prestosql.sql.planner.plan.SimplePlanRewriter$RewriteContext.lambda$defaultRewrite$0(SimplePlanRewriter.java:73)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:195)
	at java.base/java.util.Collections$2.tryAdvance(Collections.java:4747)
```

**Comment by veysiertekin** [comment]
We also have experienced same behaviour at one of the tables that contain `NaN` values. even if we filter out by `is_nan(field) = false`

As a workaround, temporary table without `NaN` values works just fine.

**Comment by findepi** [comment]
@veysiertekin this can be the same problem as reported by @martint or similar (but different).
Do you have a full stacktrace?

**Comment by veysiertekin** [comment]
Hi @findepi ,

Here is the stacktrace (line numbers may change because of Presto 331):

```
Caused by: java.sql.SQLException: Query failed (#20200630_093207_06823_kiqec): Error opening Hive split <a path to parquet> (offset=0, length=33554432): cannot use Double NaN as range bound
	at io.prestosql.jdbc.PrestoResultSet.resultsException(PrestoResultSet.java:1839)
	at io.prestosql.jdbc.PrestoResultSet$ResultsPageIterator.computeNext(PrestoResultSet.java:1819)
	at io.prestosql.jdbc.PrestoResultSet$ResultsPageIterator.computeNext(PrestoResultSet.java:1758)
	at io.prestosql.jdbc.$internal.guava.collect.AbstractIterator.tryToComputeNext(AbstractIterator.java:141)
	at io.prestosql.jdbc.$internal.guava.collect.AbstractIterator.hasNext(AbstractIterator.java:136)
	at io.prestosql.jdbc.$internal.guava.collect.TransformedIterator.hasNext(TransformedIterator.java:42)
	at io.prestosql.jdbc.$internal.guava.collect.Iterators$ConcatenatedIterator.getTopMetaIterator(Iterators.java:1311)
	at io.prestosql.jdbc.$internal.guava.collect.Iterators$ConcatenatedIterator.hasNext(Iterators.java:1327)
	at io.prestosql.jdbc.LengthLimitedIterator.hasNext(LengthLimitedIterator.java:42)
	at io.prestosql.jdbc.PrestoResultSet.next(PrestoResultSet.java:143)
	at org.jkiss.dbeaver.model.impl.jdbc.exec.JDBCResultSetImpl.next(JDBCResultSetImpl.java:268)
	at org.jkiss.dbeaver.model.impl.jdbc.exec.JDBCResultSetImpl.nextRow(JDBCResultSetImpl.java:176)
	... 13 more
Caused by: io.prestosql.spi.PrestoException: Error opening Hive split <a path to parquet> (offset=0, length=33554432): cannot use Double NaN as range bound
	at io.prestosql.plugin.hive.parquet.ParquetPageSourceFactory.createParquetPageSource(ParquetPageSourceFactory.java:252)
	at io.prestosql.plugin.hive.parquet.ParquetPageSourceFactory.createPageSource(ParquetPageSourceFactory.java:130)
	at io.prestosql.plugin.hive.HivePageSourceProvider.createHivePageSource(HivePageSourceProvider.java:190)
	at io.prestosql.plugin.hive.HivePageSourceProvider.createPageSource(HivePageSourceProvider.java:108)
	at io.prestosql.plugin.hive.HivePageSourceProvider.createPageSource(HivePageSourceProvider.java:91)
	at io.prestosql.plugin.base.classloader.ClassLoaderSafeConnectorPageSourceProvider.createPageSource(ClassLoaderSafeConnectorPageSourceProvider.java:49)
	at io.prestosql.split.PageSourceManager.createPageSource(PageSourceManager.java:61)
	at io.prestosql.operator.ScanFilterAndProjectOperator$SplitToPages.process(ScanFilterAndProjectOperator.java:243)
	at io.prestosql.operator.ScanFilterAndProjectOperator$SplitToPages.process(ScanFilterAndProjectOperator.java:175)
	at io.prestosql.operator.WorkProcessorUtils$3.process(WorkProcessorUtils.java:319)
	at io.prestosql.operator.WorkProcessorUtils$ProcessWorkProcessor.process(WorkProcessorUtils.java:372)
	at io.prestosql.operator.WorkProcessorUtils$3.process(WorkProcessorUtils.java:306)
	at io.prestosql.operator.WorkProcessorUtils$ProcessWorkProcessor.process(WorkProcessorUtils.java:372)
	at io.prestosql.operator.WorkProcessorUtils$3.process(WorkProcessorUtils.java:306)
	at io.prestosql.operator.WorkProcessorUtils$ProcessWorkProcessor.process(WorkProcessorUtils.java:372)
	at io.prestosql.operator.WorkProcessorUtils.getNextState(WorkProcessorUtils.java:221)
	at io.prestosql.operator.WorkProcessorUtils.lambda$processStateMonitor$2(WorkProcessorUtils.java:200)
	at io.prestosql.operator.WorkProcessorUtils$ProcessWorkProcessor.process(WorkProcessorUtils.java:372)
	at io.prestosql.operator.WorkProcessorUtils.getNextState(WorkProcessorUtils.java:221)
	at io.prestosql.operator.WorkProcessorUtils.lambda$finishWhen$3(WorkProcessorUtils.java:215)
	at io.prestosql.operator.WorkProcessorUtils$ProcessWorkProcessor.process(WorkProcessorUtils.java:372)
	at io.prestosql.operator.WorkProcessorSourceOperatorAdapter.getOutput(WorkProcessorSourceOperatorAdapter.java:148)
	at io.prestosql.operator.Driver.processInternal(Driver.java:379)
	at io.prestosql.operator.Driver.lambda$processFor$8(Driver.java:283)
	at io.prestosql.operator.Driver.tryWithLock(Driver.java:675)
	at io.prestosql.operator.Driver.processFor(Driver.java:276)
	at io.prestosql.execution.SqlTaskExecution$DriverSplitRunner.processFor(SqlTaskExecution.java:1075)
	at io.prestosql.execution.executor.PrioritizedSplitRunner.process(PrioritizedSplitRunner.java:163)
	at io.prestosql.execution.executor.TaskExecutor$TaskRunner.run(TaskExecutor.java:484)
	at io.prestosql.$gen.Presto_327_764_g5ee56cf____20200623_193347_2.run(Unknown Source)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.lang.IllegalArgumentException: cannot use Double NaN as range bound
	at io.prestosql.spi.predicate.Marker.<init>(Marker.java:79)
	at io.prestosql.spi.predicate.Marker.create(Marker.java:88)
	at io.prestosql.spi.predicate.Marker.exactly(Marker.java:114)
	at io.prestosql.spi.predicate.Range.equal(Range.java:84)
	at io.prestosql.spi.predicate.SortedRangeSet.of(SortedRangeSet.java:80)
	at io.prestosql.spi.predicate.ValueSet.of(ValueSet.java:60)
	at io.prestosql.spi.predicate.Domain.singleValue(Domain.java:84)
	at io.prestosql.parquet.predicate.TupleDomainParquetPredicate.getDomain(TupleDomainParquetPredicate.java:281)
	at io.prestosql.parquet.predicate.TupleDomainParquetPredicate.effectivePredicateMatches(TupleDomainParquetPredicate.java:123)
	at io.prestosql.parquet.predicate.TupleDomainParquetPredicate.matches(TupleDomainParquetPredicate.java:118)
	at io.prestosql.parquet.predicate.PredicateUtils.dictionaryPredicatesMatch(PredicateUtils.java:122)
	at io.prestosql.parquet.predicate.PredicateUtils.predicateMatches(PredicateUtils.java:95)
	at io.prestosql.plugin.hive.parquet.ParquetPageSourceFactory.createParquetPageSource(ParquetPageSourceFactory.java:201)
	... 32 more
```

**Comment by findepi** [comment]
@veysiertekin so this bug is different than what @martint reported.
This issue is going to be fixed by #4266, but that won't help you yet.

When working on that fix, i realized the potential problem in ORC and Parquet
and created #4267 for that. I now updated that issue with the stacktrace
you provided. Please follow that issue.
